### ISSUE TITLE
[ProblemDetails] Regression between 4.1.2 and 4.2.2

### ISSUE THREAD

**Issue by nulltoken** [stoplightio/prism#1830]
## Describe the bug

While upgrading from 4.1.2 to 4.2.2 some tests started to fail on our side. It looks like something regressed Prism side with regards to (at least) validation of "date-time" format.

## To Reproduce

- *Remove all node_modules folder* (Important as we want to rollback deps to their previous versions)
- Baseline
  - Checkout https://github.com/nulltoken/prism/tree/ntk/date-time-4.1.2 branch
  - yarn
  - yarn jest functional.spec.ts -t "HttpValidator validateInput\(\) all validations are turned on properly validate date-time format"
  - This should pass
- Regression
  - Checkout https://github.com/nulltoken/prism/tree/ntk/date-time-4.2.2 branch
  - yarn
  - yarn jest functional.spec.ts -t "HttpValidator validateInput\(\) all validations are turned on properly validate date-time format"
  - This should fail with the following error

```text
 ● HttpValidator › validateInput() › all validations are turned on › properly validate date-time format ("2021-02-18T12:02:16.49")

    expect(received).toMatchSnapshot()

    Snapshot name: `HttpValidator validateInput() all validations are turned on properly validate date-time format ("2021-02-18T12:02:16.49") 1`        

    - Snapshot  - 11
    + Received  +  9

      Object {
    -   "_tag": "Left",
    -   "left": Array [
    -     Object {
    -       "code": "format",
    -       "message": "should match format \"date-time\"",
    -       "path": Array [
    -         "query",
    -         "updated_since",
    -       ],
    -       "severity": 0,
    +   "_tag": "Right",
    +   "right": Object {
    +     "method": "get",
    +     "url": Object {
    +       "path": "/todos",
    +       "query": Object {
    +         "updated_since": "2021-02-18T12:02:16.49",
    +       },
    +     },
        },
    -   ],
      }
```

It seems that 4.2.2 now considers `2021-02-18T12:02:16.49` as a valid "date-time" (whereas it doesn't specify any timezone)

## Expected behavior

Ajv-formats seems to be pretty explicit with "date-time" requiring a timezone part (cf. https://github.com/ajv-validator/ajv-formats#formats)

Prism should keep on reporting this not matching query parameter as a problem.

/cc @P0lip

**Comment by nulltoken** [comment]
@philsturgeon :wave: When do you think yhis could get triaged?

**Comment by philsturgeon** [comment]
@lottamus do you think your team could take a swing at this? Seems like it could be telling of a larger issue, like maybe we're not validating formats anymore, or something.

**Comment by nulltoken** [comment]
@lottamus Do you need any more data from me?

**Comment by chohmann** [comment]
@nulltoken this issue has been brought into our current sprint, so we'll start looking into this!

**Comment by chohmann** [comment]
@nulltoken we're using AJV Formats and they are incorrectly returning that the time is valid without a timezone even though they say it's required. I've pinpointed the issue in their code to:

https://github.com/ajv-validator/ajv-formats/blob/master/src/formats.ts#L154

Gonna try to put up a PR to fix this in AJV Formats. Once that's merged and released, we can up the version in Prism.

**Comment by chohmann** [comment]
This fix is included in Prism 4.3.1!

**Comment by nulltoken** [comment]
@chohmann 😍
